src/denormalizer/getOne.js

```diff
@@ -26,13 +30,23 @@ export function getOne(one, state, schema = '') {
     throw new Error('State argument is invalid, should be an object.');
   }
 
-  const resolvedSchema = resolveSchemaName(one, schema);
+  const resolvedSchema = resolveReferenceSchemaType(one, schema);
 
-  const schemaPaths = rio.schemaPaths;
+  const schemaPaths = rio.resourcePaths;
   if (!schemaPaths[resolvedSchema]) {
     throw new Error(`Storage for resolved schema ${resolvedSchema} doesn't exists in state.`);
   }
-  const storageMap = createSchemasMap(state, schemaPaths);
+
+  // TODO: use memoization https://www.npmjs.com/package/mem, https://www.npmjs.com/package/memoizee
+  if (
+    state !== lastState ||
+    schemaPaths !== lastSchemaPaths ||
+    !storageMap
+  ) {
```

Comment:
This logic is repeated in multiple functions, can you extract it as a helper function?